src/main/kotlin/com/github/michaelbull/result/bind/BindFailure.kt

```diff
@@ -0,0 +1,9 @@
+package com.github.michaelbull.result.bind
+
+sealed class NoStackTraceException : Exception() {
+    override fun fillInStackTrace(): Throwable {
+        return this
+    }
+
+    object BindFailure : NoStackTraceException()
+}
```

Comment:
Is there a reason we model the `NoStackTraceException` as a `sealed class` where the only impl is `BindFailure`?

I suspect this can be simplified down to:

```kotlin
object BindFailure : Exception() {
    override fun fillInStackTrace(): Throwable {
        return this
    }
}
```

Reply:
no reason at all. I think at the time of creatign it I just wanted to be verbose about what everything was. But ye i think it makes sense to simplify it!